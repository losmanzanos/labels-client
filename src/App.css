.App {
  text-align: center;
  display: flex;
  flex-direction: column;
}

h2 {
  padding-bottom: 10px;
  padding-top: 30px;
}
