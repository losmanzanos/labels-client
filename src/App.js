import { Switch, Route } from "react-router-dom";

import "./App.css";
import Upload from "./components/upload/Upload";
import Header from "./components/header/header";

import Error from "./components/error/Error";
// import EditEntry from "../EditEntry/EditEntry";
import About from "./components/About/About";
import Info from "./components/Info/Info";

import PublicOnlyRoute from "./components/Utils/PublicOnlyRoute";
import PrivateRoute from "./components/Utils/PrivateRoute";

import LoginPage from "../src/routes/LoginPage/LoginPage";
import RegistrationPage from "../src/routes/RegistrationPage/RegistrationPage";
import NotFoundPage from "../src/routes/NotFoundPage/NotFoundPage";
import { AppProvider } from "./contexts/AppContext";

function App() {
  return (
    <AppProvider>
      <Error>
        <div className="App">
          <Header />

          <Switch>
            <Route exact path="/" component={About} />
            <Route path="/about" component={Info} />
            {/* <Route path="/gratitude/:id" component={EditEntry} /> */}
            <PublicOnlyRoute path="/login" component={LoginPage} />
            <PublicOnlyRoute path="/register" component={RegistrationPage} />
            <PrivateRoute path="/gallery" component={Upload} />
            <Route component={NotFoundPage} />
          </Switch>

          {/* <Upload /> */}
        </div>
      </Error>
    </AppProvider>
  );
}

export default App;
