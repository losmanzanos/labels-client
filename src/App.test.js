import renderer from "react-test-renderer";

//Smoke Test

// make React available
import React from "react";

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";

// make the App component available
import App from "./App";

// this is the test case
it("renders without crashing", () => {
  // first create a DOM element to render the component into
  const div = document.createElement("div");

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
    div
  );

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

//Snapshot

it("renders the App as expected", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
