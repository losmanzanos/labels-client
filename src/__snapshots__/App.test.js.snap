// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App as expected 1`] = `
<div
  className="App"
>
  <nav
    className="Header"
  >
    <div
      className="Harold"
    >
      <h1>
        Harold
      </h1>
      <div>
        <img
          alt="Harold"
          id="Harold"
          src="Harold.jpg"
        />
      </div>
    </div>
    <div
      className="Header__not-logged-in"
    >
      <a
        aria-current="page"
        className="hover active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
       | 
      <a
        aria-current={null}
        className="hover"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
       | 
      <a
        aria-current={null}
        className="hover"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
       | 
      <a
        aria-current={null}
        className="hover"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </nav>
  <div
    className="About"
  >
    <p>
      Struggling to figure out what that picture you just downloaded is? Not hip enough to understand the latest memes?
    </p>
    <div>
      <img
        alt="Granny"
        id="Granny"
        src="granny.jpeg"
      />
    </div>
    <p>
      If this is you, Harold is here to help! Harold doesn't have the best vision, but he'll do his best to help you out!
    </p>
    <p
      id="smaller"
    >
      (Or maybe he'll just confuse you even more...
      <br />
      If that's the case, sorry.)
    </p>
  </div>
  <div
    id="contact"
  >
    <p>
      Contact me at:
      <br />
      <a
        href="mailto:dahc.reverse@gmail.com?subject=YOU%20ROCK!"
      >
        dahc.reverse@gmail.com
      </a>
    </p>
  </div>
</div>
`;
